staging/vhost-device-can/src/vhu_can.rs

```diff
@@ -483,42 +501,71 @@ impl VhostUserCanBackend {
         }
 
         for desc_chain in requests {
-            let descriptors: Vec<_> = desc_chain.clone().collect();
-
-            if descriptors.len() != 2 {
-                return Err(Error::UnexpectedDescriptorCount(descriptors.len()));
-            }
+            let atomic_mem = self.mem.as_ref().unwrap().memory();
 
-            let desc_request = descriptors[0];
-            if desc_request.is_write_only() {
+            let Ok(mut reader) = desc_chain.clone().reader(&atomic_mem) else {
+                trace!("Error::UnexpectedWriteOnlyDescriptor");
                 return Err(Error::UnexpectedWriteOnlyDescriptor(0));
-            }
+            };
 
-            if (desc_request.len() as usize) > size_of::<VirtioCanFrame>()
-                || (desc_request.len() as usize) < (size_of::<VirtioCanFrame>() - 64)
-            {
+            let Ok(mut writer) = desc_chain.clone().writer(&atomic_mem) else {
+                trace!("Error::UnexpectedReadableDescriptor");
+                return Err(Error::UnexpectedReadableDescriptor(0));
+            };
+
+            let reader_len = reader.available_bytes();
+            if reader_len < size_of::<VirtioCanHeader>() {
```

Comment:
I think you do not need to check the available bytes at this point. You can just fail when you do a `read_obj` later.

Reply:
Thanks, the available_bytes check has been removed, I have adjust the code to fail at the write/read phase.